sonarr/series.go

```diff
@@ -27,50 +105,52 @@ func (s *Sonarr) GetSeries(tvdbID int64) ([]*Series, error) {
 }
 
 func (s *Sonarr) GetSeriesContext(ctx context.Context, tvdbID int64) ([]*Series, error) {
-	params := make(url.Values)
+	var output []*Series
 
+	params := make(url.Values)
 	if tvdbID != 0 {
 		params.Add("tvdbId", strconv.FormatInt(tvdbID, starr.Base10))
 	}
 
-	var series []*Series
-
-	_, err := s.GetInto(ctx, "v3/series", params, &series)
-	if err != nil {
+	if _, err := s.GetInto(ctx, bpSeries, params, &output); err != nil {
 		return nil, fmt.Errorf("api.Get(series): %w", err)
 	}
 
-	return series, nil
+	return output, nil
 }
 
 // UpdateSeries updates a series in place.
-func (s *Sonarr) UpdateSeries(seriesID int64, series *Series) error {
-	return s.UpdateSeriesContext(context.Background(), seriesID, series)
+func (s *Sonarr) UpdateSeries(series *Series) (*Series, error) {
+	return s.UpdateSeriesContext(context.Background(), series)
 }
 
-func (s *Sonarr) UpdateSeriesContext(ctx context.Context, seriesID int64, series *Series) error {
+func (s *Sonarr) UpdateSeriesContext(ctx context.Context, series *Series) (*Series, error) {
+	var output *Series
+
 	params := make(url.Values)
 	params.Add("moveFiles", "true")
 
 	var body bytes.Buffer
 	if err := json.NewEncoder(&body).Encode(series); err != nil {
-		return fmt.Errorf("json.Marshal(series): %w", err)
+		return nil, fmt.Errorf("json.Marshal(series): %w", err)
 	}
 
-	_, err := s.Put(ctx, "v3/series/"+strconv.FormatInt(seriesID, starr.Base10), params, &body)
-	if err != nil {
-		return fmt.Errorf("api.Put(series): %w", err)
+	uri := path.Join(bpSeries, strconv.Itoa(int(series.ID)))
+	if _, err := s.PutInto(ctx, uri, params, &body, &output); err != nil {
+		return nil, fmt.Errorf("api.Put(series): %w", err)
 	}
 
-	return nil
+	return output, nil
 }
 
 // AddSeries adds a new series to Sonarr.
-func (s *Sonarr) AddSeries(series *AddSeriesInput) (*AddSeriesOutput, error) {
+func (s *Sonarr) AddSeries(series *Series) (*Series, error) {
```

Comment:
I'm not sure this is a change I want to do. Did you test this against Sonarr?

Reply:
Yeah I looked more. These two structs are wildly different and will inevitably lead a confused user. Adding a series requires like two files, and this change makes it look like it's dozens. This may make the code cleaner, but it does not make the user experience better imho.

Reply:
This also adds things like `AddSeriesOptions` to the output, which is just more confusion.

Reply:
To be honest, I wasn't sure about the `AddSeriesOption` as well, I'll make few more tests to be sure about it, and maybe find a better approach.
However, I don't think the two structs are much different. Actually, I think that they are mainly the same apart from `AddSeriesOption`. To create a working series you need at least 8 parameters, and many others can be configured, I still haven't checked all of them, but I'll try few things before coming back to you.
One of the biggest advantage of using the same struct is that you could take the output of a lookup and pass it directly to the AddSeries.